ibis/backends/dask/trace.py

```diff
@@ -0,0 +1,147 @@
+import functools
+import logging
+import traceback
+from datetime import datetime
+
+from ibis.config import get_option, set_option
+from ibis.expr import types as ir
+
+from .dispatcher import TwoLevelDispatcher
+
+"""Module that adds tracing to dask execution.
+With tracing enabled, this module will log time and call stack information of
+the executed expression. Call stack information is presented with indentation
+level.
+For example:
+import dask.dataframe as dd
+import pandas as pd
+import logging
+import ibis.expr.datatypes as dt
+import ibis.dask
+from ibis.udf.vectorized import elementwise
+from ibis.dask import trace
+logging.basicConfig()
+trace.enable()
+df = dd.from_pandas(
+    pd.DataFrame({'a': [1, 2, 3]})
+)
+con = ibis.dask.connect({"table1": df})
+@elementwise(
+    input_type=[dt.double],
+    output_type=dt.double
+)
+def add_one(v):
+    import time
+    time.sleep(5)
+    return v + 1
+table = con.table("table1")
+table = table.mutate(b=add_one(table['a']))
+table.execute()
+Output:
+DEBUG:ibis.dask.trace: main_execute Selection
```

Comment:
I don't think the tracing make sense for dask if this is lazily executed...

Reply:
I think it still helps to trace what ibis is doing?

Reply:
Yeah that's true. The pandas tracing tool was original written to debug performance, I guess you can use it for just getting the stack trace but it's less useful that way.

Either way, I would prefer not to duplicate all the tracing logics, but we can address as follow up.

Reply:
Yep - I believe I can import most of it from pandas.